### PR TITLE
docs: trusted publishing support

### PR DESCRIPTION
**Pull request by Hofer-Julian**
This needs https://github.com/prefix-dev/rattler-build-action/pull/51 in order to actually test the workflow

### CODE REVIEW

File: docs/authentication_and_upload.md [hunk 1]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
 
 ### prefix.dev
 
-To upload to [prefix.dev](https://prefix.dev), you need to have an account and a
-token. You can create a token in the settings of your account. The token is used
+#### OpenID Connect (OIDC)
+
+`rattler-build` also supports OIDC with GitHub and prefix.dev.
```
Comment (baszalmstra) [inline docs/authentication_and_upload.md:61]:
```suggestion
`rattler-build` supports authentication with https://prefix.dev through OIDC with GitHub Actions.
```

File: docs/authentication_and_upload.md [hunk 2]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
...
+That way you don't need to manually retrieve and store a token.
```
Comment (baszalmstra) [inline docs/authentication_and_upload.md:62]:
```suggestion
An API key is no longer required, rattler-build can manage the complete authentication workflow for you.
```

File: docs/authentication_and_upload.md [hunk 3]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
...
+        shell: bash
+        run: |
+          shopt -s nullglob
+          EXIT_CODE=0
+          for pkg in $(find output -type f \( -name "*.conda" -o -name "*.tar.bz2" \) ); do
+            if ! rattler-build upload prefix -c my-channel "$file"; then
+              EXIT_CODE=1
+            fi
+          done
+          exit $EXIT_CODE
```
Comment (baszalmstra) [inline docs/authentication_and_upload.md:90]:
```suggestion
      - name: Upload all packages
        run: rattler-build upload prefix -c my-channel *.conda *.tar.bz2
```

Comment (baszalmstra) [reply docs/authentication_and_upload.md:90]: Not sure if that works..

File: docs/authentication_and_upload.md [hunk 4]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
...
```
Comment (tdejager) [inline docs/authentication_and_upload.md:90]: As discussed maybe we could wait for `rattler-build` action to be updated to make this nicer?

File: docs/authentication_and_upload.md [hunk 5]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
 
 ### prefix.dev
 
-To upload to [prefix.dev](https://prefix.dev), you need to have an account and a
-token. You can create a token in the settings of your account. The token is used
+#### OpenID Connect (OIDC)
```
Comment (wolfv) [inline docs/authentication_and_upload.md:59]: Can we start calling it something like Trusted publishing via OIDC or so?

OIDC is very technical and trusted publishing is maybe easier to grasp (also if you know pypi of course).

**Comment by Hofer-Julian** [comment]
I've now tested it and it works as expected

**Review by wolfv — CHANGES_REQUESTED** [review]

File: docs/authentication_and_upload.md [hunk 6]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
...
-To upload to [prefix.dev](https://prefix.dev), you need to have an account and a
-token. You can create a token in the settings of your account. The token is used
+#### Trusted publishing via OIDC
+
+`rattler-build` supports authentication with https://prefix.dev through OIDC with GitHub Actions.
+An API key is no longer required, rattler-build can manage the complete authentication workflow for you.
+You only have to set up a specific repository and workflow under "Trusted Publishers" on prefix.dev.
+
+Here you can find an example GitHub Actions workflow
+
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply docs/authentication_and_upload.md:69]: Done

File: docs/authentication_and_upload.md [hunk 7]
```diff
@@ -56,8 +56,45 @@ authenticate with the server.
...
+```yaml
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply docs/authentication_and_upload.md]: Added the title. I assume that's what you meant :)

**Review by wolfv — APPROVED** [review]

**Pull request merged**